Unable to set up local environment

## Before reporting this bug:
- [x] Verify there is not an existing issue for it already
- [x] You are using the latest version of `toqito` with a python version `>=3.10`

## Describe the bug

When running `uv sync` in the project root I get an error complaining about he absence of `umfpack.h`:

<details>
<summary>full build log</summary>
<pre>
uv sync
Using CPython 3.13.0
Creating virtual environment at: .venv
Resolved 142 packages in 13ms
      Built toqito @ file:///Users/nate/code/toqito
      Built picos==2.6.2
  × Failed to build `cvxopt==1.3.2`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `setuptools.build_meta.build_wheel` failed (exit status: 1)

      [stdout]
      running bdist_wheel
      running build
      running build_py
      creating build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      copying src/python/misc.py -> build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      copying src/python/_version.py -> build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      copying src/python/msk.py -> build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      copying src/python/__init__.py -> build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      copying src/python/solvers.py -> build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      copying src/python/cvxprog.py -> build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      copying src/python/modeling.py -> build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      copying src/python/info.py -> build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      copying src/python/coneprog.py -> build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      copying src/python/printing.py -> build/lib.macosx-11.0-arm64-cpython-313/cvxopt
      running build_ext
      building 'base' extension
      creating build/temp.macosx-11.0-arm64-cpython-313/src/C
      cc -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -arch arm64 -mmacosx-version-min=11.0 -Wno-nullability-completeness
      -Wno-expansion-to-defined -Wno-undef-prefix -fPIC -Werror=unguarded-availability-new -I/Users/nate/.cache/uv/builds-v0/.tmphHhENc/include
      -I/Users/nate/.local/share/uv/python/cpython-3.13.0-macos-aarch64-none/include/python3.13 -c src/C/base.c -o
      build/temp.macosx-11.0-arm64-cpython-313/src/C/base.o
      cc -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -arch arm64 -mmacosx-version-min=11.0 -Wno-nullability-completeness
      -Wno-expansion-to-defined -Wno-undef-prefix -fPIC -Werror=unguarded-availability-new -I/Users/nate/.cache/uv/builds-v0/.tmphHhENc/include
      -I/Users/nate/.local/share/uv/python/cpython-3.13.0-macos-aarch64-none/include/python3.13 -c src/C/dense.c -o
      build/temp.macosx-11.0-arm64-cpython-313/src/C/dense.o
      cc -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -arch arm64 -mmacosx-version-min=11.0 -Wno-nullability-completeness
      -Wno-expansion-to-defined -Wno-undef-prefix -fPIC -Werror=unguarded-availability-new -I/Users/nate/.cache/uv/builds-v0/.tmphHhENc/include
      -I/Users/nate/.local/share/uv/python/cpython-3.13.0-macos-aarch64-none/include/python3.13 -c src/C/sparse.c -o
      build/temp.macosx-11.0-arm64-cpython-313/src/C/sparse.o
      cc -bundle -undefined dynamic_lookup -arch arm64 -mmacosx-version-min=11.0 -LModules/_hacl build/temp.macosx-11.0-arm64-cpython-313/src/C/base.o
      build/temp.macosx-11.0-arm64-cpython-313/src/C/dense.o build/temp.macosx-11.0-arm64-cpython-313/src/C/sparse.o
      -L/usr/lib -L/Users/nate/.local/share/uv/python/cpython-3.13.0-macos-aarch64-none/lib -lm -llapack -lblas -o
      build/lib.macosx-11.0-arm64-cpython-313/cvxopt/base.cpython-313-darwin.so
      building 'blas' extension
      cc -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -arch arm64 -mmacosx-version-min=11.0 -Wno-nullability-completeness
      -Wno-expansion-to-defined -Wno-undef-prefix -fPIC -Werror=unguarded-availability-new -I/Users/nate/.cache/uv/builds-v0/.tmphHhENc/include
      -I/Users/nate/.local/share/uv/python/cpython-3.13.0-macos-aarch64-none/include/python3.13 -c src/C/blas.c -o
      build/temp.macosx-11.0-arm64-cpython-313/src/C/blas.o
      cc -bundle -undefined dynamic_lookup -arch arm64 -mmacosx-version-min=11.0 -LModules/_hacl build/temp.macosx-11.0-arm64-cpython-313/src/C/blas.o -L/usr/lib
      -L/Users/nate/.local/share/uv/python/cpython-3.13.0-macos-aarch64-none/lib -lblas -o build/lib.macosx-11.0-arm64-cpython-313/cvxopt/blas.cpython-313-darwin.so
      building 'lapack' extension
      cc -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -arch arm64 -mmacosx-version-min=11.0 -Wno-nullability-completeness
      -Wno-expansion-to-defined -Wno-undef-prefix -fPIC -Werror=unguarded-availability-new -I/Users/nate/.cache/uv/builds-v0/.tmphHhENc/include
      -I/Users/nate/.local/share/uv/python/cpython-3.13.0-macos-aarch64-none/include/python3.13 -c src/C/lapack.c -o
      build/temp.macosx-11.0-arm64-cpython-313/src/C/lapack.o
      cc -bundle -undefined dynamic_lookup -arch arm64 -mmacosx-version-min=11.0 -LModules/_hacl build/temp.macosx-11.0-arm64-cpython-313/src/C/lapack.o
      -L/usr/lib -L/Users/nate/.local/share/uv/python/cpython-3.13.0-macos-aarch64-none/lib -llapack -lblas -o
      build/lib.macosx-11.0-arm64-cpython-313/cvxopt/lapack.cpython-313-darwin.so
      building 'umfpack' extension
      cc -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -arch arm64 -mmacosx-version-min=11.0 -Wno-nullability-completeness
      -Wno-expansion-to-defined -Wno-undef-prefix -fPIC -Werror=unguarded-availability-new -I/usr/local/include -I/Users/nate/.cache/uv/builds-v0/.tmphHhENc/include
      -I/Users/nate/.local/share/uv/python/cpython-3.13.0-macos-aarch64-none/include/python3.13 -c src/C/umfpack.c -o
      build/temp.macosx-11.0-arm64-cpython-313/src/C/umfpack.o

      [stderr]
      /Users/nate/.cache/uv/builds-v0/.tmphHhENc/lib/python3.13/site-packages/setuptools/dist.py:765: SetuptoolsDeprecationWarning: License classifiers are
      deprecated.
      !!

              ********************************************************************************
              Please consider removing the following classifiers in favor of a SPDX license expression:

              License :: OSI Approved :: GNU General Public License v3 (GPLv3)

              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************

      !!
        self._finalize_license_expression()
      ld: warning: search path 'Modules/_hacl' not found
      ld: warning: search path 'Modules/_hacl' not found
      ld: warning: search path 'Modules/_hacl' not found
      src/C/umfpack.c:23:10: fatal error: 'umfpack.h' file not found
      #include "umfpack.h"
               ^~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/cc' failed with exit code 1

      hint: This error likely indicates that you need to install a library that provides "umfpack.h" for `cvxopt@1.3.2`
  help: `cvxopt` (v1.3.2) was included because `toqito` (v1.1.3) depends on `picos` (v2.6.2) which depends on `cvxopt`
</pre>
</details>

I found [this website](https://people.engr.tamu.edu/davis/suitesparse.html) to download the tool that seems to contain umfpack, but I'm not sure how to actually install it. If this is a necessary step it should be documented.

## System Information where `toqito` is installed

MacOS 14.8.3

--- a/.github/workflows/benchmark.yml
+++ b/.github/workflows/benchmark.yml
@@ -125,15 +125,15 @@ jobs:
       with:
         python-version: ${{ env.PYTHON_VERSION }}
     
-    - name: Install Poetry
-      run: pip install poetry
-    
+    - name: Install uv
+      uses: astral-sh/setup-uv@v5
+
     - name: Setup toqito environment
       run: |
         cd toqito
         echo "Setting up toqito development environment..."
-        poetry install --with dev
-        poetry add pytest-benchmark --group dev pytest-memray sympy pygal
+        uv sync --group dev
+        uv pip install pytest-benchmark pytest-memray pygal
         echo "Environment setup complete"
     
     - name: Verify baseline benchmark exists
@@ -187,7 +187,7 @@ jobs:
 
         cd ../toqito
 
-        PYTEST_CMD="poetry run pytest ../toqito-bench/scripts/benchmark_toqito.py"
+        PYTEST_CMD="uv run pytest ../toqito-bench/scripts/benchmark_toqito.py"
         if [ -n "$FILTER" ]; then
           PYTEST_CMD="$PYTEST_CMD -k \"$FILTER\""
         fi
@@ -240,7 +240,7 @@ jobs:
 
         cd ../toqito
 
-        PYTEST_CMD="poetry run pytest ../toqito-bench/scripts/benchmark_toqito.py"
+        PYTEST_CMD="uv run pytest ../toqito-bench/scripts/benchmark_toqito.py"
         if [ -n "$FILTER" ]; then
           PYTEST_CMD="$PYTEST_CMD -k \"$FILTER\""
         fi

--- a/.github/workflows/build-test-actions.yml
+++ b/.github/workflows/build-test-actions.yml
@@ -20,7 +20,7 @@ jobs:
       fail-fast: false
       matrix:
           # Add versions here to expand support matrix
-        python-version: ["3.10", "3.11", "3.12"]
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
         os: [ubuntu-latest, macos-latest]
     runs-on: ${{ matrix.os }}
     env:
@@ -39,11 +39,15 @@ jobs:
         if: runner.os == 'Linux'
         run: |
           sudo apt-get install -y libblas-dev liblapack-dev
+      - name: Install suite-sparse for macOS
+        if: runner.os == 'macOS'
+        run: |
+          brew install suite-sparse
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
       - name: Install toqito + dependencies
         run: |
-          python -m pip install --upgrade pip
-          pip install poetry
-          poetry install --without docs --no-interaction
+          uv sync --group dev --no-group docs --no-group lint
 
 
   toqito-pytest:
@@ -51,7 +55,7 @@ jobs:
       fail-fast: false
       matrix:
         # Add versions here to expand support matrix
-        python-version: ["3.10", "3.11", "3.12"]
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
         os: [ubuntu-latest, macos-latest]
     runs-on: ${{ matrix.os }}
         # https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy
@@ -67,17 +71,17 @@ jobs:
         if: runner.os == 'Linux'
         run: |
           sudo apt-get install -y libblas-dev liblapack-dev
-      - name: Install blas, lapack for macOS
+      - name: Install blas, lapack, suite-sparse for macOS
         if: runner.os == 'macOS'
         run: |
-          brew install openblas lapack
+          brew install openblas lapack suite-sparse
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
       - name: Install toqito + dependencies
         run: |
-          python -m pip install --upgrade pip
-          pip install poetry
-          poetry install --with dev --without docs,lint --no-interaction
+          uv sync --group dev --no-group docs --no-group lint
       - name: Run tests
-        run: poetry run pytest -v --cov-config=.coveragerc --cov-report=xml --cov-report term-missing:skip-covered --cov=toqito
+        run: uv run pytest -v --cov-config=.coveragerc --cov-report=xml --cov-report term-missing:skip-covered --cov=toqito
       - name: Upload coverage information
         uses: codecov/codecov-action@v5
         with:
@@ -87,5 +91,5 @@ jobs:
           files: ./coverage.xml
           verbose: true
       - name: Build artifacts
-        run: poetry build
+        run: uv build
 

--- a/.github/workflows/publish-release.yml
+++ b/.github/workflows/publish-release.yml
@@ -22,15 +22,13 @@ jobs:
         uses: actions/setup-python@v6
         with:
           python-version: "3.11"
-      - name: Install Python dependencies
-        run: |
-          python -m pip install --upgrade pip
-          make install requirements
-          pip install setuptools wheel twine
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
       - name: Build and publish
         env:
           TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
           TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
         run: |
-          python setup.py sdist bdist_wheel
+          uv build
+          pip install twine
           twine upload dist/*  

--- a/docs/contributing.rst
+++ b/docs/contributing.rst
@@ -17,7 +17,7 @@ We welcome contributions from external contributors, and this document describes
 .. warning::
      Avoid ad-hoc :code:`pip install -e .` workflows; the project standardizes on :code:`uv` for syncing dependencies.
 
-4. As stated in :ref:`getting_started_reference-label`, ensure you have Python 3.10 to 3.12 installed on your machine or in 
+4. As stated in :ref:`getting_started_reference-label`, ensure you have Python 3.10 or greater installed on your machine or in
    a virtual environment (`pyenv <https://github.com/pyenv/pyenv>`_, `pyenv tutorial <https://realpython.com/intro-to-pyenv/>`_).
    Consider using a `virtual environment <https://docs.python.org/3/tutorial/venv.html>`_.
    You can also use :code:`pyenv` with :code:`virtualenv` `to manage different Python

--- a/docs/getting_started.rst
+++ b/docs/getting_started.rst
@@ -11,11 +11,17 @@ Getting started
 Installing
 ----------
 
-1. Ensure you have Python 3.10 or greater installed on your machine or in 
-a virtual environment (`pyenv <https://github.com/pyenv/pyenv>`_, `pyenv tutorial <https://realpython.com/intro-to-pyenv/>`_). 
+1. Ensure you have Python 3.10 or greater installed on your machine or in
+a virtual environment (`pyenv <https://github.com/pyenv/pyenv>`_, `pyenv tutorial <https://realpython.com/intro-to-pyenv/>`_).
 
-.. warning::
-    Python 3.13 is not yet supported due to incompatibilities with some dependencies (e.g., :code:`cvxopt`).
+.. note::
+    On macOS, the :code:`cvxopt` dependency (pulled in by :code:`picos`) may need to be built from source
+    if a pre-built wheel is not available for your Python version and macOS version. If you encounter a
+    build error mentioning :code:`umfpack.h`, install the required system library with:
+
+    .. code-block:: bash
+
+        brew install suite-sparse
 
 2. Consider using a `virtual environment <https://docs.python.org/3/tutorial/venv.html>`_.
 You can also use :code:`pyenv` with :code:`virtualenv` `to manage different Python versions <https://github.com/pyenv/pyenv-virtualenv>`_. 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -21,6 +21,7 @@ classifiers = [
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
     "License :: OSI Approved :: BSD License",
     "Operating System :: OS Independent",
 ]

